mailpoet/lib/Automation/Integrations/WooCommerce/Fields/OrderFieldsFactory.php

```diff
@@ -0,0 +1,347 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Integrations\WooCommerce\Fields;
+
+use DateTimeImmutable;
+use MailPoet\Automation\Engine\Data\Field;
+use MailPoet\Automation\Engine\WordPress;
+use MailPoet\Automation\Integrations\WooCommerce\Payloads\OrderPayload;
+use MailPoet\Automation\Integrations\WooCommerce\WooCommerce;
+use WC_Order;
+use WC_Order_Item_Product;
+use WC_Payment_Gateway;
+use WC_Product;
+use WP_Post;
+
+class OrderFieldsFactory {
+  /** @var TermOptionsBuilder */
+  private $termOptionsBuilder;
+
+  /** @var WordPress */
+  private $wordPress;
+
+  /** @var WooCommerce */
+  private $wooCommerce;
+
+  public function __construct(
+    TermOptionsBuilder $termOptionsBuilder,
+    WordPress $wordPress,
+    WooCommerce $wooCommerce
+  ) {
+    $this->termOptionsBuilder = $termOptionsBuilder;
+    $this->wordPress = $wordPress;
+    $this->wooCommerce = $wooCommerce;
+  }
+
+  /** @return Field[] */
+  public function getFields(): array {
+    return array_merge(
+      [
+        new Field(
+          'woocommerce:order:billing-company',
+          Field::TYPE_STRING,
+          __('Billing company', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_billing_company();
+          }
+        ),
+        new Field(
+          'woocommerce:order:billing-phone',
+          Field::TYPE_STRING,
+          __('Billing phone', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_billing_phone();
+          }
+        ),
+        new Field(
+          'woocommerce:order:billing-city',
+          Field::TYPE_STRING,
+          __('Billing city', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_billing_city();
+          }
+        ),
+        new Field(
+          'woocommerce:order:billing-postcode',
+          Field::TYPE_STRING,
+          __('Billing postcode', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_billing_postcode();
+          }
+        ),
+        new Field(
+          'woocommerce:order:billing-state',
+          Field::TYPE_STRING,
+          __('Billing state/county', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_billing_state();
+          }
+        ),
+        new Field(
+          'woocommerce:order:billing-country',
+          Field::TYPE_STRING,
```

Comment:
Would `ENUM´ make sense here?

Reply:
@websupporter You mean using something like `WC_Countries`, right? Sounds like a good idea 👍

Reply:
https://github.com/mailpoet/mailpoet/pull/4984/commits/e0938bd6bb7a1452f3875602733c3ecea05a850b + https://github.com/mailpoet/mailpoet/pull/4984/commits/30b7eb6c57a91dd6a6804a2180824a99c744fa87